src/ooni/dns_injection_impl.hpp

```diff
@@ -37,7 +37,7 @@ class DNSInjectionImpl : public DNSTestImpl {
     void main(std::string input, Settings options,
               std::function<void(json)> &&cb) {
         entry["injected"] = nullptr;
-        query("A", "IN", input, options["dns/nameserver"],
+        query("A", "IN", input, options["backend"],
```

Comment:
Here I changed the name used for the option because what we are actually setting here is the backend for the test (i.e., "dns/nameserver" is an lower-level option).